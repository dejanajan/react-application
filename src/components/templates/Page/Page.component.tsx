import React from "react";
import * as PageStyle from "./Page.styles";

interface PageType {
  children: React.ReactNode;
}

export function PageContainer(props: PageType) {
  const { children } = props;

  return <PageStyle.Page>{children}</PageStyle.Page>;
}

export default PageContainer;