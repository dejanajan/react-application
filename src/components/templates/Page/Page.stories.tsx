import React from "react";
import { storiesOf } from "@storybook/react";
import PageContainer from "./Page.component";



storiesOf("PageContainer", module).add("default", () => (
  <PageContainer>
    <p>Page content</p>
  </PageContainer>
));
