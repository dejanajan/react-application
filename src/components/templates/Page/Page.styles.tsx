
import styled from "styled-components"

export const Page = styled.main`
    margin: 20px;
    display: flex;
    flex-direction: column;
`
