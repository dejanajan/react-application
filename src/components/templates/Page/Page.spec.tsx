import React from "react";
import Page from "./Page.component";
import renderer from "react-test-renderer";

test("Render Page", () => {
  const page = renderer
  .create(
    <Page>
      <p>Test this page content</p>
    </Page>
  );

  const tree = page.toJSON();
  expect(tree).toMatchSnapshot();
});
