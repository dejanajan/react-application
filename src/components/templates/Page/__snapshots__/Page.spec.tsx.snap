// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Page 1`] = `
<main
  className="sc-AxjAm cvhCLc"
>
  <p>
    Test this page content
  </p>
</main>
`;
